engine/table/src/main/java/io/deephaven/engine/table/impl/TimeTable.java

```diff
@@ -100,28 +143,25 @@ private void refresh(final boolean notifyListeners) {
         try {
             final DateTime dateTime = timeProvider.currentTime();
             long rangeStart = lastIndex + 1;
-            if (columnSource.baseTime == null) {
+            if (columnSource.startTime == null) {
                 lastIndex = 0;
-                columnSource.baseTime = new DateTime(
+                columnSource.startTime = new DateTime(
                         LongNumericPrimitives.lowerBin(dateTime.getNanos(), columnSource.period));
             } else {
                 lastIndex = Math.max(lastIndex,
-                        DateTimeUtils.minus(dateTime, columnSource.baseTime) / columnSource.period);
+                        DateTimeUtils.minus(dateTime, columnSource.startTime) / columnSource.period);
             }
 
             if (rangeStart <= lastIndex) {
-                // If we have a period longer than 10s, print out that the timetable has been updated. This can be
-                // useful when analyzing what's gone wrong in the logs. It is capped at periods of 5s, so we don't
-                // end up with too much log spam for short interval time tables. 5s is not so coincidentally the period
-                // of the Jvm Heap: messages.
-                if (columnSource.period >= 5_000_000_000L) {
-                    log.info().append("TimeTable updated to ").append(columnSource.get(lastIndex).toString()).endl();
+                final RowSet addedRange = RowSetFactory.fromRange(rangeStart, lastIndex);
+                final RowSet removedRange = isStreamTable && rangeStart > 0
```

Comment:
As you noticed, we need to remove all prior rows if `isStreamTable` unconditionally.